swagger/v1/api_spec.json

```diff
@@ -587,21 +636,60 @@
           }
         }
       },
-      "ECFParticipantAction": {
-        "description": "The details of a participant",
+      "ECFParticipantChangeSchedule": {
+        "description": "An ECF participant change schedule action",
         "type": "object",
-        "required": [
-          "type",
-          "attributes"
-        ],
         "properties": {
-          "type": {
-            "description": "The data type",
+          "schedule_identifier": {
+            "description": "The new schedule of the participant",
             "type": "string",
-            "example": "participant"
+            "enum": [
+              "ecf-september-standard-2021",
+              "ecf-september-extended-2021"
+            ],
+            "example": "career-break"
           },
-          "attributes": {
-            "$ref": "#/components/schemas/ECFParticipantWithdrawal"
+          "course_identifier": {
+            "description": "The type of course the participant is enrolled in",
+            "type": "string",
+            "enum": [
+              "ecf-induction",
+              "ecf-mentor"
+            ],
+            "example": "ecf-induction"
+          }
+        },
+        "required": [
+          "schedule_identifier",
+          "course_identifier"
+        ],
+        "example": {
+          "schedule_identifier": "ecf-september-extended-2021",
+          "course_identifier": "ecf-mentor"
+        }
+      },
+      "ECFParticipantChangeScheduleRequest": {
+        "description": "The change schedule request for a participant",
+        "type": "object",
+        "required": [
+          "data"
+        ],
+        "properties": {
+          "data": {
```

Comment:
Does `data` not need to have a `type` and `description` too to make this a valid schema?

Reply:
Interesting, it seems to work fine, but I added them.